drivers/sensor/lps2xdf/lps2xdf.c

```diff
@@ -0,0 +1,490 @@
+
+	/* Set bdu and if_inc recommended for driver usage */
+	if (lps22df_init_set(ctx, LPS22DF_DRV_RDY) < 0) {
+		LOG_ERR("%s: Not able to set device to ready state", dev->name);
+		return -EIO;
+	}
+
+	if (ON_I3C_BUS(cfg)) {
+		lps22df_bus_mode_t bus_mode;
+
+		/* Select bus interface */
+		bus_mode.filter = LPS22DF_AUTO;
+		bus_mode.interface = LPS22DF_SEL_BY_HW;
+		lps22df_bus_mode_set(ctx, &bus_mode);
+	}
+
+	/* set sensor default odr */
+	LOG_DBG("%s: odr: %d", dev->name, cfg->odr);
+	ret = lps22df_mode_set_odr_raw(dev, cfg->odr);
+	if (ret < 0) {
+		LOG_ERR("%s: Failed to set odr %d", dev->name, cfg->odr);
+		return ret;
+	}
+
+#ifdef CONFIG_LPS2XDF_TRIGGER
+	if (lps2xdf_init_interrupt(dev, DEVICE_VARIANT_LPS22DF) < 0) {
+		LOG_ERR("Failed to initialize interrupt.");
+		return -EIO;
+	}
+#endif
+
+	return 0;
+}
+#endif
+
+#if DT_HAS_COMPAT_STATUS_OKAY(st_lps28dfw)
+static int st_lps28dfw_init(const struct device *dev)
+{
+	const struct lps2xdf_config *const cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lps28dfw_id_t id;
+	lps28dfw_stat_t status;
+	uint8_t tries = 10;
+	int ret;
+
+#if DT_HAS_COMPAT_ON_BUS_STATUS_OKAY(st_lps28dfw, i3c)
+	if (cfg->i3c.bus != NULL) {
+		struct lps2xdf_data *data = dev->data;
+		/*
+		 * Need to grab the pointer to the I3C device descriptor
+		 * before we can talk to the sensor.
+		 */
+		data->i3c_dev = i3c_device_find(cfg->i3c.bus, &cfg->i3c.dev_id);
+		if (data->i3c_dev == NULL) {
+			LOG_ERR("Cannot find I3C device descriptor");
+			return -ENODEV;
+		}
+	}
+#endif
+
+	if (lps28dfw_id_get(ctx, &id) < 0) {
+		LOG_ERR("%s: Not able to read dev id", dev->name);
+		return -EIO;
+	}
+
+	if (id.whoami != LPS28DFW_ID) {
+		LOG_ERR("%s: Invalid chip ID 0x%02x", dev->name, id.whoami);
+		return -EIO;
+	}
+
+	LOG_DBG("%s: chip id 0x%x", dev->name, id.whoami);
+
+	/* Restore default configuration */
+	if (lps28dfw_init_set(ctx, LPS28DFW_RESET) < 0) {
+		LOG_ERR("%s: Not able to reset device", dev->name);
+		return -EIO;
+	}
+
+	do {
+		if (!--tries) {
+			LOG_DBG("sw reset timed out");
+			return -ETIMEDOUT;
+		}
+		k_usleep(LPS2XDF_SWRESET_WAIT_TIME_US);
+
+		if (lps28dfw_status_get(ctx, &status) < 0) {
+			return -EIO;
+		}
+	} while (status.sw_reset);
+
+	/* Set bdu and if_inc recommended for driver usage */
+	if (lps28dfw_init_set(ctx, LPS28DFW_DRV_RDY) < 0) {
+		LOG_ERR("%s: Not able to set device to ready state", dev->name);
+		return -EIO;
+	}
+
+	if (ON_I3C_BUS(cfg)) {
+		lps28dfw_bus_mode_t bus_mode;
+
+		/* Select bus interface */
+		bus_mode.filter = LPS28DFW_AUTO;
+		bus_mode.interface = LPS28DFW_SEL_BY_HW;
+		lps28dfw_bus_mode_set(ctx, &bus_mode);
+	}
+
+	/* set sensor default odr */
+	LOG_DBG("%s: odr: %d", dev->name, cfg->odr);
+	ret = lps28dfw_mode_set_odr_raw(dev, cfg->odr);
+	if (ret < 0) {
+		LOG_ERR("%s: Failed to set odr %d", dev->name, cfg->odr);
+		return ret;
+	}
+
+#ifdef CONFIG_LPS2XDF_TRIGGER
+	if (lps2xdf_init_interrupt(dev, DEVICE_VARIANT_LPS28DFW) < 0) {
```

Comment:
Actually I think this code is wrong. It was wrong also in lps22df (my bad). I'm starting to review the code for all ST sensors and since we are changing this one we can do the change on thi PR itself. Basically here the problem is that lps22df can be multi instantiated, with one instance with triggers and one with not. So, we do not have to rely on ```CONFIG_LPS2XDF_TRIGGER```, but instead on the DT (```DT_INST_NODE_HAS_PROP``` for drdy_gpios). See lsm6dsv16x implementation, where .trig_enabled is set to '1' if either one of int gpios is defined in DT.

Reply:
Right, I did not see that. It's changed now.